test/e2e/kcp_upgrade.go

```diff
@@ -32,6 +32,8 @@ import (
 )
 
 // KCPUpgradeSpecInput is the input for KCPUpgradeSpec.
+// Deprecated: This struct has been deprecated  and will be removed in a future release.
+// Please use ClusterUpgradeConformanceSpec and ClusterUpgradeConformanceSpecInput instead.
 type KCPUpgradeSpecInput struct {
```

Comment:
> The test module, clusterctl, and experiments do not provide any backward compatible guarantees.

Let's remove these methods and add a note in https://cluster-api.sigs.k8s.io/developer/providers/v1.0-to-v1.1.html

Reply:
Perfect, will do